xdrv_21_wemo.ino

```diff
@@ -0,0 +1,486 @@
+    "<name>MetaInfo</name>"
+    "<dataType>string</dataType>"
+    "<defaultValue>0</defaultValue>"
+    "</stateVariable>"
+    "</serviceStateTable>"
+    "</scpd>\r\n\r\n";
+
+const char WEMO_RESPONSE_STATE_SOAP[] PROGMEM =
+    "<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\" s:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">"
+    "<s:Body>"
+    "<u:%cetBinaryStateResponse xmlns:u=\"urn:Belkin:service:basicevent:1\">"
+    "<BinaryState>%d</BinaryState>"
+    "</u:%cetBinaryStateResponse>"
+    "</s:Body>"
+    "</s:Envelope>\r\n";
+
+const char WEMO_SETUP_XML[] PROGMEM =
+    "<?xml version=\"1.0\"?>"
+    "<root xmlns=\"urn:Belkin:device-1-0\">"
+    "<device>"
+    "<deviceType>urn:Belkin:device:controllee:1</deviceType>"
+    "<friendlyName>{x1</friendlyName>"
+    "<manufacturer>Belkin International Inc.</manufacturer>"
+    "<modelName>Socket</modelName>"
+    "<modelNumber>3.1415</modelNumber>"
+    "<UDN>uuid:{x2</UDN>"
+    "<serialNumber>{x3</serialNumber>"
+    "<binaryState>0</binaryState>"
+    "<serviceList>"
+    "<service>"
+    "<serviceType>urn:Belkin:service:basicevent:1</serviceType>"
+    "<serviceId>urn:Belkin:serviceId:basicevent1</serviceId>"
+    "<controlURL>/upnp/control/basicevent1</controlURL>"
+    "<eventSubURL>/upnp/event/basicevent1</eventSubURL>"
+    "<SCPDURL>/eventservice.xml</SCPDURL>"
+    "</service>"
+    "<service>"
+    "<serviceType>urn:Belkin:service:metainfo:1</serviceType>"
+    "<serviceId>urn:Belkin:serviceId:metainfo1</serviceId>"
+    "<controlURL>/upnp/control/metainfo1</controlURL>"
+    "<eventSubURL>/upnp/event/metainfo1</eventSubURL>"
+    "<SCPDURL>/metainfoservice.xml</SCPDURL>"
+    "</service>"
+    "</serviceList>"
+    "</device>"
+    "</root>\r\n";
+#endif
+
+/********************************************************************************************/
+
+class WemoSwitch
+{
+
+private:
+  ESP8266WebServer *_webServer = NULL;
+  int _localPort = 80;
+  int _deviceId;
+
+  String WemoSerialnumber(void)
+  {
+    char serial[18];
+
+    snprintf_P(serial, sizeof(serial), PSTR("201612K%08X-%d"), ESP_getChipId(), _deviceId);
+    return String(serial);
+  }
+
+  String WemoUuid(void)
+  {
+    char uuid[29];
+
+    snprintf_P(uuid, sizeof(uuid), PSTR("Socket-1_0-%s"), WemoSerialnumber().c_str());
+    return String(uuid);
+  }
+
+public:
+  WemoSwitch(int deviceId, ESP8266WebServer *webServer)
+  {
+    _deviceId = deviceId;
+    _webServer = webServer;
+
+    AddLog_P2(LOG_LEVEL_DEBUG, PSTR("\nDEVICE #%d LISTENING ON PORT %d"), _deviceId, _localPort);
+  }
+  WemoSwitch(int deviceId, int localPort)
+  {
+    _deviceId = deviceId;
+    _localPort = localPort;
+
+    AddLog_P2(LOG_LEVEL_DEBUG, PSTR("\nDEVICE #%d LISTENING ON PORT %d"), _deviceId, _localPort);
+
+    _webServer = new ESP8266WebServer(_localPort);
+
+    _webServer->begin();
+  }
+
+  ~WemoSwitch()
+  {
+    if (_deviceId > 1)
+    {
+      _webServer->close();
```

Comment:
You never call `delete _webServer` but I believe that this destructor is never called anyways.